File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
    dfs(root);
    return root;
    }
    TreeNode* dfs(TreeNode* node){
        if(!node) return nullptr;
        TreeNode* left= dfs(node->left);
        TreeNode* right= dfs(node->right);
        TreeNode* dummy=node->left;
        node->left=node->right;
        node->right=dummy;
        return node;

    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if root is None:
            return 0
        res=0
        def dfs(node,maxSoFar):
            nonlocal res
            if node is None:
                return
            if node.val>=maxSoFar:
                res+=1
            newMax=max(node.val,maxSoFar)
            dfs(node.left,newMax)
            dfs(node.right,newMax)
        
        dfs(root,root.val)
        return res
        